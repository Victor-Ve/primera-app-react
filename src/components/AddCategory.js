import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategorias }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategorias((categoria) => [inputValue, ...categoria]);
      setInputValue('');
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        className="indigo lighten-5"
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategorias: PropTypes.func.isRequired,
};

export default AddCategory;
