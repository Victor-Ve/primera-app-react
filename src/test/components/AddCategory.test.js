import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import AddCategory from '../../components/AddCategory';

describe('Pruebas en <AddCatagory />', () => {
  const setCategory = jest.fn();
  let wrapper = shallow(<AddCategory setCategorias={setCategory} />);

  //realizar antes de cada prueba
  beforeEach(() => {
    jest.clearAllMocks(); //limpiar simulaciones
    wrapper = shallow(<AddCategory setCategorias={setCategory} />);
  });

  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'hola mundo';

    input.simulate('change', { target: { value } });

    expect(wrapper.find('p').text().trim()).toBe(value);
  });

  test('No debe de postear la informacion con submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategory).not.toHaveBeenCalled();
  });

  test('debe de llamar el setCategory y limpiar la caja de texto', () => {
    const value = 'Hola mundo';
    wrapper.find('input').simulate('change', { target: { value } });

    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategory).toHaveBeenCalled();
    // expect(setCategory).toHaveBeenCalledTimes(1); otra forma de hacerlo

    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
