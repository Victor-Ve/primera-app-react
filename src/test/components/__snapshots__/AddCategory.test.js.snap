// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCatagory /> debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    className="indigo lighten-5"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
